content/en/tutorials/gitlab_ci_testcontainers/index.md

```diff
@@ -0,0 +1,431 @@
+used for different runners.
+
+{{<command>}}
+$ docker volume create gitlab-runner-config
+<disable-copy>
+gitlab-runner-config
+</disable-copy>
+{{</command>}}
+
+The agent that will interact with the system to create the runner needs to be run:
+
+{{<command>}}
+$ docker run -d --name gitlab-runner \
+--restart always \
+-v /var/run/docker.sock:/var/run/docker.sock \
+-v gitlab-runner-config:/etc/gitlab-runner/ \
+gitlab/gitlab-runner:latest
+<disable-copy>
+79ad150847bd78fc567b08cafd96fc6bfec8f1946feb2beea89d7c6c395c01c4
+</disable-copy>
+{{</command>}}
+
+The breakdown for this command:
+- The container is named `gitlab-runner`.
+- It is configured to always restart.
+- It has access to the Docker socket on the host machine, allowing it to manage Docker containers - which is very important here.
+- It uses the named volume previously defined for persistent configuration storage.
+- It uses the latest GitLab Runner image.
+
+The next step is to register the runner:
+
+{{<command>}}
+$ docker run --rm -it \
+-v gitlab-runner-config:/etc/gitlab-runner/ \
+gitlab/gitlab-runner:latest register
+<disable-copy>
+79ad150847bd78fc567b08cafd96fc6bfec8f1946feb2beea89d7c6c395c01c4
+</disable-copy>
+{{</command>}}
+
+Follow the instructions that are prompted:
+
+{{< figure src="runner-config.png" width="80%" height="auto">}}
+
+In the container logs you should see this:
+
+```commandline
+Configuration loaded                                builds=0 max_builds=1
+```
+
+Let's look at the `config.toml` file and make the final adjustment before successfully running the pipeline.
+For running a job that does not require any additional containers to be created, you can stop here. However, since 
+we need to run Docker commands in our CI/CD jobs, we must configure GitLab Runner to support those commands. 
+This method requires `privileged` mode.
+
+Let's use the current running container to do that. Run the following:
+
+```commandline
+$ docker exec -it gitlab-runner bin/bash
+```
+
+Inside the container, let's run:
+
+{{<command>}}
+$ cd etc/gitlab-runner
+$ ls -al
+<disable-copy>
+total 24
+drwx------ 3 root root 4096 May 16 19:58 .
+drwxr-xr-x 1 root root 4096 May 16 19:56 ..
+-rw------- 1 root root   14 May 16 19:50 .runner_system_id
+drwx------ 2 root root 4096 May  3 17:36 certs
+-rwx------ 1 root root  781 May 16 19:52 config.toml
+</disable-copy>
+{{</command>}}
+
+You can see the file that contains all the runner configurations.
+
+{{<command>}}
+$ apt update && apt install nano
+$ nano config.toml
+{{</command>}}
+
+The `privileged` field needs to be changed to `true`. Now the configurations should look like this:
+
+```toml
+connection_max_age = "15m0s"
+shutdown_timeout = 0
+
+[session_server]
+  session_timeout = 1800
+
+[[runners]]
+  name = "localstack-testcontainers-runner"
+  url = "https://gitlab.com"
+  id = 36509569
+  token = "glrt-1SLQUHi3TjsEi23Ro48e"
```

Comment:
Is it a real token? 👀

Reply:
it's a real token, but the runner does not exist anymore, so there's nothing you can do with it anymore. but I'll replace it with a generic pattern and explain it